src/Storages/Hive/StorageHive.cpp

```diff
@@ -0,0 +1,667 @@
+                         .can_be_true)
+                {
+                    LOG_TRACE(log, "Skip split {} of hive file {}", i, file->getPath());
+                    skip_splits.insert(i);
+                }
+            }
+            file->setSkipSplits(skip_splits);
+        }
+
+        {
+            std::lock_guard lock{hive_files_mutex};
+            hive_files.push_back(file);
+        }
+    };
+
+    ThreadPool pool{num_streams};
+    if (!partitions.empty())
+    {
+        const auto partition_names = partition_name_types.getNames();
+        const auto partition_types = partition_name_types.getTypes();
+
+        for (const auto & partition : partitions)
+        {
+            auto process_partition = [&]()
+            {
+                LOG_DEBUG(log, "Process partition {}", boost::join(partition.values, ","));
+
+                /// Skip partition "__HIVE_DEFAULT_PARTITION__"
+                bool has_default_partition = false;
+                for (const auto & value : partition.values)
+                {
+                    if (value == "__HIVE_DEFAULT_PARTITION__")
+                    {
+                        has_default_partition = true;
+                        break;
+                    }
+                }
+                if (has_default_partition)
+                {
+                    return;
+                }
+
+                std::vector<Range> ranges;
+                WriteBufferFromOwnString wb;
+                if (partition.values.size() != partition_names.size())
+                    throw Exception(
+                        fmt::format("Partition value size not match, expect {}, but got {}", partition_names.size(), partition.values.size()),
+                        ErrorCodes::INVALID_PARTITION_VALUE);
+
+                for (size_t i = 0; i < partition.values.size(); ++i)
+                {
+                    if (i != 0)
+                        writeString(",", wb);
+                    writeString(partition.values[i], wb);
+                }
+                writeString("\n", wb);
+
+                ReadBufferFromString buffer(wb.str());
+                auto format = FormatFactory::instance().getInputFormat(
+                    "CSV", buffer, partition_key_expr->getSampleBlock(), getContext(), getContext()->getSettingsRef().max_block_size);
+                auto pipeline = QueryPipeline(std::move(format));
+                auto reader = std::make_unique<PullingPipelineExecutor>(pipeline);
+                Block block;
+                if (!reader->pull(block) || !block.rows())
+                    throw Exception("Could not parse partition value: " + wb.str(), ErrorCodes::INVALID_PARTITION_VALUE);
+
+                FieldVector fields(partition_names.size());
+                for (size_t i = 0; i < partition_names.size(); ++i)
+                {
+                    block.getByPosition(i).column->get(0, fields[i]);
```

Comment:
So just for this line need to do all this thing with writing partitions list into in memory buffer, then reading it into block in order to get just fields? Can we just create needed fields as is?

Reply:
> Ok, I hope this is the final iteration of review. Haven't now looked at `RemoteReadBufferCache.cpp` file yet. Will do soon.

Thank you for you long-time reviewing, we will modify code as you demand if it is reasonable as soon as possible.

Reply:
> So just for this line need to do all this thing with writing partitions list into in memory buffer, then reading it into block in order to get just fields? Can we just create needed fields as is?

@kssenii I can't think a better way to get fields from partition values, what do you advice?

Reply:
> @kssenii I can't find a better way to get fields from partition values, what do you advice?

But you just need string `Field` values? Why not just create them from strings?

Reply:
I think firstly we must decide data types of partition values, then transform string to field value using corresponding `ISerialization` of data types . But  for interface `ISerialization::deserializeTextCSV`, input is `ReadBuffer`, output is `IColumn`. This means we have to read field values from `IColumns` after `ISerialization::deserializeTextCSV`.  

Implementing in above way  is all right, but it looks not easier than current way.